sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -610,7 +611,8 @@ def __init__(
       coder=None,
       use_standard_sql=False,
       flatten_results=True,
-      kms_key=None):
+      kms_key=None,
+      use_json_exports=False):
```

Comment:
not supporting JSON would increase the amount of changes that users need to make to migrate to the new source, as the data output types would change. I don't think this would be ideal. Wouldn't it be better to allow users to migrate to the new source without having to change the rest of their pipeline?

Reply:
Assuming Avro is a superior format, I think we should drop support for json (but not right away). My preference would be to have this choice be orthogonal to the choice of types (and having the bytes default to be base64 encoded feels really wrong) so do think having a backwards compatibility parameter is preferable. (If it's just the date and bytes types, I would split this into two parameters: `use_base64_encoded_bytes` and `datedtime_type`.)

Reply:
> "having the bytes default to be base64 encoded feels really wrong" .
I think we changed bytes to be base64-encoded several years back, because Java SDK used that representation: https://github.com/apache/beam/blob/master/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java#L181-L182

Reply:
Yes, the joys of matching what was done in Java years ago for the Dataflow service API.

Reply:
Restarting this discussion after some rebasing and preparing...

I think I'd prefer to allow users to choose the export file format. This is what we do for writing to BQ in Java and Python. Allowing users to choose output type for specific column types would add overhead of (if avro and json_format_bytes: formatbytes; if avro and json_datetime_format: formatdatetime; if not avro and not json_format_bytes: formatbytes; ...).

We can discourage users from using JSON (AVRO is already the default) - and eventually stop supporting it if that makes sense. Thoughts?